applications/dashboard/models/class.permissionmodel.php

```diff
@@ -590,6 +591,7 @@ public function getJunctionPermissions($where, $junctionTable = null, $limitToSu
             $junctionColumn = $row['JunctionColumn'];
             unset($row['PermissionID'], $row['RoleID'], $row['JunctionTable'], $row['JunctionColumn'], $row['JunctionID']);
 
+            unset($juncIDs);
```

Comment:
It's a bit hard to review this without more context about what the data pollution issue was causing/vs what was previously expected out of this function, but from what I can see, the `$juncIDs` get set again just a few lines after this, so I'm not really sure if this fully solves whatever the issue is.

Reply:
We are looping through the list of `JunctionTable` variants which were only `Category` before.
Now we are implementing another table to join.

So for the _1st cycle_ of the loop let say we got condition 
```
if ($limitToDefault === false && $junctionColumn != $junctionTable.'ID') {
```
equals to **true** and went inside `if (...){ ... }` which bring up the variable `$juncIDs` with some value.

This value is used in line 659:
```
 } elseif (isset($juncIDs)) {
```
which is fine during the _1st cycle_ . But during next loop iteration:
```
if ($limitToDefault === false && $junctionColumn != $junctionTable.'ID') {
```
 can be negative. Which means by design we expect  `$juncIDs` to be not initialized. But it was during 1st loop iteration and keeps previous data. And this bring false positive (using this polluted variable value) on the condition in line 659.

To avoid this we should _reset_ variable to some initial state that will not bring false positive trigger .